npc/custom/events/mushroom_event.txt

```diff
@@ -50,9 +50,12 @@ OnMinute10:	// Start time (every hour)
 
 OnMobKilled:
 	set .Spawn, .Spawn - 1;
-	getitem .Prize, .Amount;
-	if (.Spawn) announce "[ "+strcharinfo(PC_NAME)+" ] has killed a Mushroom. There are now "+.Spawn+" Mushroom(s) left.",bc_map;
-	else {
+	if (playerattached() > 0) {
```

Comment:
Uh, it should be a ```!= 0``` not ```> 0```
The source returns 0 on failure, which means that a negative value can be valid, maybe not now but in future this can be the case, it should be a strict ```!= 0```

Reply:
ok done